org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/metrics/ReviewAnalyser.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.log4j.Logger;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.wso2.code.quality.metrics.model.Constants.GITHUB_REVIEW_API_CLOSED_STATE;
+import static com.wso2.code.quality.metrics.model.Constants.GITHUB_REVIEW_APPROVED;
+import static com.wso2.code.quality.metrics.model.Constants.GITHUB_REVIEW_COMMENTED;
+
+/**
+ * This class is used to find the reviewers of the buggy lines of code.
+ *
+ * @since 1.0.0
+ */
+
+public class ReviewAnalyser {
+    private static final Logger logger = Logger.getLogger(ReviewAnalyser.class);
+
+    // to store the reviewed and approved users of thepull requests
+    final Set<String> approvedReviewers = new HashSet<>();
+    // to store the reviewed and commented users of the pull requests
+    final Set<String> commentedReviewers = new HashSet<>();
+    //constants for filtering github API responses
+    private final GithubApiCaller githubApiCaller = new GithubApiCaller();
+    private final Gson gson = new Gson();
+
+    /**
+     * This class is used to prevent SIC_INNER_SHOULD_BE_STATIC_ANON error that comes when building with WSO2 parent
+     * pom. As suggested by the above error an static inner class is used
+     */
+    private static class ListType extends TypeToken<List<ReviewApiResponse>> {
+    }
+
+
+    /**
+     * This is used to identify the pull requests that introduce the given commit to the code base.
+     *
+     * @param authorCommits Commits which the relevant pull request no must be found
+     * @param githubToken   Github access token for accessing github API
+     */
+    public void findReviewers(Set<String> authorCommits, String githubToken) {
+        authorCommits.forEach(commitHash -> {
+            String jsonText;
+            try {
+                jsonText = githubApiCaller.callSearchIssueApi(commitHash, githubToken);
+                Map<String, Set<Integer>> prNoWithRepoName = savePrNumberAndRepoName(jsonText);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Relevant pull requests on patch " + commitHash + " with their relevant repository " +
+                            "names are successfully saved in a map.");
+                }
+                saveReviewers(prNoWithRepoName, githubToken);
+            } catch (CodeQualityMetricsException e) {
+                logger.error(e.getMessage(), e.getCause());
+            }
+        });
+    }
+
+    /**
+     * This is used to save the pull requests with their relevant repository names in a map.
+     *
+     * @param jsonText json reponse received from the github issue API
+     * @return a map of pull requests againt their repository name
+     * @throws CodeQualityMetricsException results
+     */
+    Map<String, Set<Integer>> savePrNumberAndRepoName(String jsonText) throws CodeQualityMetricsException {
```

Comment:
@kasunsiyambalapitiya : You can give private access even it is used in a test case. AFAIK, you can use java reflection to temporary change the access modifier before calling this method from the test case.